source/manual/setting-up-new-rails-app.html.md

```diff
@@ -8,140 +8,161 @@ last_reviewed_on: 2018-03-22
+require "byebug"
+require "simplecov"
 
-Add the app to the `Procfile` and assign it the next available port
-number.
+ENV["RAILS_ENV"] ||= "test"
+require File.expand_path("../../config/environment", __FILE__)
+require "rspec/rails"
 
-    myapp: govuk_setenv myapp ./run_in.sh ../myapp bundle exec rails s -p 3456
+Dir[Rails.root.join("spec/support/**/*.rb")].each { |f| require f }
+SimpleCov.start
 
-If the app is dependent on other apps, you'll also want to add it to the
-`Pinfile` and define its dependencies.
+RSpec.configure do |config|
+  config.expose_dsl_globally = false
+  config.infer_spec_type_from_file_location!
+end
+```
 
-    process :myapp => :anotherapp
+In config, replace the content of `database.yml` with the following.
 
-To be able to access your app via a browser, you need to add it to the
-hosts list in hieradata for development. See
-[govuk-puppet](https://github.com/alphagov/govuk-puppet/blob/master/docs/adding-a-new-app.md#including-the-app-on-machines)
-for more information.
+```
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  pool: 12
+  template: template0
 
-You should then be able to view it at <http://myapp.dev.gov.uk>.
+development:
+  <<: *default
+  database: myapp_development
+  url: <%= ENV["DATABASE_URL"]%>
 
-## Defining the app in Puppet
+test:
+  <<: *default
+  database: myapp_test
+  url: <%= ENV["DATABASE_URL"].try(:sub, /([-_]development)?$/, '_test')%>
 
-There's a guide explaining [how to add a new app to the Puppet
-repository](https://github.com/alphagov/govuk-puppet/blob/master/docs/adding-a-new-app.md).
+production:
+  <<: *default
+  database: myapp_production
+  url: <%= ENV["DATABASE_URL"]%>
+```
 
-Be sure to use the same port number which you assigned in the
-`Procfile`.
+In config, replace `credentials.yml.env` and `master.key` with `secrets.yml`.
 
-## Setting up DNS entries
+```
+development:
+  secret_key_base: secret
 
-If the app requires the creation of new external (publicly accessible)
-subdomains, you can do this via [the govuk-dns-config
-repo](https://github.com/alphagov/govuk-dns-config). If you need assistance
-with pointing the records at the correct boxes, talk to Reliability
-Engineering.
+test:
+  secret_key_base: secret
 
-## Adding deployment scripts
+production:
+  secret_key_base: <%= ENV['SECRET_KEY_BASE'] %>
+```
 
-After the app has been configured and defined in Puppet, the deployment
-scripts are the next step to getting the app running on the boxes.
-Deployment scripts are bespoke to each application, but they live
-together in the [govuk-app-deployment
-repository](https://github.com/alphagov/govuk-app-deployment).
+In config, replace the content of `routes.rb` with the following healthcheck.
 
-All Rails apps use [Capistrano](http://www.capistranorb.com/) for deployment, and
-common deployment steps have been extracted into the `recipes` directory.
+```
+Rails.application.routes.draw do
+  get "/healthcheck", to: proc { [200, {}, ["OK"]] }
```

Comment:
Is there a GovukHealthcheck thing we can use instead of the default proc? That work should have superseded the need for the empty proc (I hope)

Reply:
I think that repo has been archived. This was what the script repo was doing, while any more advanced check would have to be a custom API, I reckon.